luskan/src/main/java/org/opencadc/luskan/ws/StorageTapService.java

```diff
@@ -148,38 +155,85 @@ public AvailabilityStatus getStatus() {
             CheckResource cr = new CheckDataSource("jdbc/tapuser", TAP_TEST);
             cr.check();
 
-            LocalAuthority localAuthority = new LocalAuthority();
-            RegistryClient reg = new RegistryClient();
-
             // TODO - These do not work for intTest unless the testing environment deploys these services.
-            URI credURI = localAuthority.getServiceURI(Standards.CRED_PROXY_10.toString());
-            String url = reg.getServiceURL(credURI, Standards.VOSI_AVAILABILITY, AuthMethod.ANON).toExternalForm();
+            String url = getServiceUrl(Standards.CRED_PROXY_10, Standards.VOSI_AVAILABILITY, AuthMethod.ANON);
             CheckResource checkResource = new CheckWebService(url);
             //checkResource.check();
 
-            URI usersURI = localAuthority.getServiceURI(Standards.UMS_USERS_01.toString());
-            url = reg.getServiceURL(usersURI, Standards.VOSI_AVAILABILITY, AuthMethod.ANON).toExternalForm();
+            url = getServiceUrl(Standards.UMS_USERS_01, Standards.VOSI_AVAILABILITY, AuthMethod.ANON);
             checkResource = new CheckWebService(url);
             //checkResource.check();
 
-            URI groupsURI = localAuthority.getServiceURI(Standards.GMS_SEARCH_01.toString());
-            url = reg.getServiceURL(groupsURI, Standards.VOSI_AVAILABILITY, AuthMethod.ANON).toExternalForm();
+            url = getServiceUrl(Standards.GMS_SEARCH_01, Standards.VOSI_AVAILABILITY, AuthMethod.ANON);
             checkResource = new CheckWebService(url);
             checkResource.check();
 
         } catch (CheckException ce) {
             // tests determined that the resource is not working
             isGood = false;
             note = ce.getMessage();
-        } catch (Throwable t) {
+        } catch (Throwable throwable) {
             // the test itself failed
-            log.error("test failed", t);
+            StringBuilder sb = new StringBuilder();
+            sb.append("test failed because: ");
+            while (throwable != null) {
+                sb.append(throwable.getMessage()).append(": ");
+                throwable = throwable.getCause();
+            }
+            log.error(sb.toString());
             isGood = false;
-            note = "test failed, reason: " + t;
+            note = sb.toString();
         }
         return new AvailabilityStatus(isGood, null, null, null, note);
     }
 
+    private String getServiceUrl(URI standardID, URI standard, AuthMethod authMethod)
```

Comment:
This is a vague method name with vague argument names for a very specialised bit of logic. How about `getAvailabilityForLocal(URI standardID)`. Note that LocalAuthority is a special way to lookup the local provider of some feature where it is inherent that there is one instance (so just A&A, not things like "the local TAP service").

Reply:
Renamed and updated the method